src/mame/drivers/8080bw.cpp

```diff
@@ -1318,31 +1318,83 @@ void _8080bw_state::cosmicmo(machine_config &config)
 /*                                                     */
 /*******************************************************/
 
+void _8080bw_state::invasion_io_map(address_map &map)
+{
+	map(0x00, 0x00).portr("IN0");
+	map(0x01, 0x01).portr("IN1");
+	map(0x02, 0x02).portr("IN2");
+	map(0x03, 0x03).portr("IN3");
+	// That's probably wrong. Sidam Invasion uses a totally redesigned hardware layout. To be done later...
+	map(0x03, 0x03).w("soundboard", FUNC(invaders_audio_device::p1_w));
+	map(0x05, 0x05).w("soundboard", FUNC(invaders_audio_device::p2_w));
+	map(0x06, 0x06).w(m_watchdog, FUNC(watchdog_timer_device::reset_w));
+}
+
+
 static INPUT_PORTS_START( invasion )
 	PORT_INCLUDE( sicv )
 
+	PORT_MODIFY("IN0")
+	PORT_BIT( 0x70, IP_ACTIVE_HIGH, IPT_CUSTOM ) PORT_CUSTOM_MEMBER(mw8080bw_state, invaders_in1_control_r)  
+	// SW1:5-8 Unused but mapped to port 0 in hw.
+	PORT_DIPUNUSED_DIPLOC( 0x80, 0x00, "SW1:5" )
+	PORT_DIPUNUSED_DIPLOC( 0x07, 0x00, "SW1:6,7,8" )
```

Comment:
You need to use individual items for each DIP switch.  Doing it like this will make the three DIP switches toggle as a single item – they should be possible to toggle individually.

Reply:
Fixed. I've not yet pushed the code because I need an advice to resolve the "DIP settings" question you arised below...